### PR TITLE
KEP review 3 changes

### PR DESCRIPTION
**Pull request by jeffvance**

### CODE REVIEW

File: keps/sig-storage/20191125-bucket-provisioning.md [hunk 1]
```diff
@@ -77,9 +79,10 @@ File and block are first class citizens within the Kubernetes ecosystem.  Object
 +  _BucketClass_ - A cluster-scoped custom resource containing fields defining the provisioner and an immutable parameter set.
    + _In Greenfield_: an abstraction of new bucket provisioning.
    + _In Brownfield_: an abstration of an existing objet store bucket.
-+ _BucketContent_ - A cluster-scoped custom resource, bound to a Bucket and containing relevant metadata. 
++ _BucketContent_ - A cluster-scoped custom resource bound to a Bucket and containing relevant metadata.
 + _Container Object Storage Interface (COSI)_ -  A specification of gRPC data and methods making up the communication protocol between the driver and the sidecar.
 + _COSI Controller_ - A central controller responsible for mananing Buckets, BucketContents, and Secrets cluster-wide.
++ _COSIDriver_ - A cluster-scoped custom resource which serves the purpose of registering a driver. It is created by the sidecar controller and guarantees unique driver names. Unlike the _CSIDriver_, it is not optional.
```
Comment (copejon) [inline keps/sig-storage/20191125-bucket-provisioning.md]: per side-bar convo - let's rename this to COSIRegistration.  Reason:  there is no overlap in purpose between this object and a CSIDriver.

Additionally, CSI has punted on solving driver name collisions, which I think reduces the need to solve it here.

File: keps/sig-storage/20191125-bucket-provisioning.md [hunk 2]
```diff
@@ -188,21 +216,19 @@ metadata:
...
-  generateBucketName: [4]
-  bucketClassName: [5]
-  secretName: [6]
-  protocol: [7]
-	accessMode: {"ro", "rw"} [8]
+  bucketPrefix: [3]
+  bucketClassName: [4]
+  secretName: [5]
+  protocol: [6]
+	accessMode: {"ro", "rw"} [7]
```
Comment (copejon) [inline keps/sig-storage/20191125-bucket-provisioning.md]: This tab is failing the ./hack/verify.sh scripts.  should be converted to single spaces.

File: keps/sig-storage/20191125-bucket-provisioning.md [hunk 3]
```diff
@@ -188,21 +216,19 @@ metadata:
...
-  bucketContentName: [9]
+  bucketContentName: [8]
   phase:
   conditions: 
 ```
 1. `labels`: COSI controller adds the label to its managed resources to easy CLI GET ops.  Value is the driver name returned by GetDriverInfo() rpc*.
 1. `finalizers`: COSI controller adds the finalizer to defer Bucket deletion until backend deletion ops succeed.
-1. `bucketName`: Desired name of the bucket to be created**.
-1. `generateBucketName`: Desired prefix to a randomly generated bucket name. Mutually exclusive with `bucketName`**.
+1. `bucketPrefix`: prefix prepended to a randomly generated bucket name. API validation _may_ ensure that `Bucket` namespace + `bucketPrefix` + generated name is less than the max length of a metadata.name.
```
Comment (copejon) [inline keps/sig-storage/20191125-bucket-provisioning.md]: > `Bucket` namespace + `bucketPrefix` + generated name

Using the namespace is probably overkill if we're generating a random suffix anyway.

Comment (jeffvance) [reply keps/sig-storage/20191125-bucket-provisioning.md]: That was a mistake on my part. I confused the actual bucket name with our proposed name of the `BucketContent` object. Will remove.

File: keps/sig-storage/20191125-bucket-provisioning.md [hunk 4]
```diff
@@ -281,5 +309,23 @@ parameters: string:string [6]
...
     - `Delete`:  the bucket and its contents are destroyed
     - `Retain`:  the bucket and its contents are preserved, only the user’s access privileges are terminated
+    - `Reuse` : TBD
+    - `Erase`: TBD
 - `bucketContentName`: (Optional). An admin defined BucketContent representing a Brownfield or Static bucket.  A non-nil value in this field prevents the BucketClass from being used for Greenfield.
-- `parameters`: object store specific key-value pairs passed to the driver.
\ No newline at end of file
+- `parameters`: object store specific key-value pairs passed to the driver.
+
+#### COSIDriver
```
Comment (copejon) [inline keps/sig-storage/20191125-bucket-provisioning.md]: COSIDriverRegistration?

Comment (copejon) [reply keps/sig-storage/20191125-bucket-provisioning.md]: Also, is this something we want to include or continue the dialogue before committing to an API?

Comment (jeffvance) [reply keps/sig-storage/20191125-bucket-provisioning.md]: I'd say lets keep it in for now (so there's a description in writing) and call it out in the next review mtg.

**Pull request merged**